tests/unit_tests/sql_lab_test.py

```diff
@@ -125,6 +127,128 @@ def test_execute_sql_statement_with_rls(
     SupersetResultSet.assert_called_with([(42,)], cursor.description, db_engine_spec)
 
 
+@mock.patch.dict(
+    "superset.sql_lab.config",
+    {"SQLLAB_PAYLOAD_MAX_MB": 50},  # Set the desired config value for testing
+)
+def test_execute_sql_statement_exceeds_payload_limit(mocker: MockerFixture) -> None:
+    """
+    Test for `execute_sql_statements` when the result payload size exceeds the limit,
+    and check if the correct log message is captured without raising the exception.
+    """
+
+    sql_statement = "SELECT 42 AS answer"
+    query_id = 1
```

Comment:
nit: as these aren't reused anywhere, and are fairly simple, I'd almost just inline them directly on lines 176-177